esp/bindings/SOAP/Platform/soapbind.cpp

```diff
@@ -258,7 +258,7 @@ int CHttpSoapBinding::HandleSoapRequest(CHttpRequest* request, CHttpResponse* re
             response->sendBasicChallenge(m_challenge_realm.str(), false);
         else if (status == SOAP_AUTHENTICATION_ERROR)
         {
-            throw MakeStringExceptionDirect(401,"Unauthorized Access");
+            throw makeStringExceptionV(401, "Unauthorized Access: %s - user=%s", ctx->queryServiceName(nullptr), ctx->queryUserId());
```

Comment:
There is potential when running a test ESP (without login) that queryUserId() will be null.

It's so weird that queryServiceName(const char *) takes a string as input.  No idea how or why that happened.

Reply:
> It's so weird that queryServiceName(const char *) takes a string as input. No idea how or why that happened.

yeah, never used, so passing nullptr to it is fine for now, but wasn't sure.

>There is potential when running a test ESP (without login) that queryUserId() will be null.

right, I will change to guard against it.

Reply:
@afishbeckln - see 2nd commit.